executor/tests/test_executor.py

```diff
@@ -15,11 +15,19 @@
 import cook.config as cc
 import cook.executor as ce
 import cook.io_helper as cio
-from tests.utils import assert_message, assert_status, cleanup_output, close_sys_outputs, ensure_directory, \
-    get_random_task_id, parse_message, redirect_stderr_to_file, redirect_stdout_to_file, wait_for, \
+from tests.utils import assert_message, assert_messages, assert_status, cleanup_output, close_sys_outputs, \
+    ensure_directory, get_random_task_id, parse_message, redirect_stderr_to_file, redirect_stdout_to_file, wait_for, \
     FakeExecutorConfig, FakeMesosExecutorDriver
 
 
+def sleep_and_set_stop_signal(stop_signal, wait_seconds):
+    """Busy waits for wait_seconds seconds before setting stop_signal."""
+    for _ in range(wait_seconds):
+        if not stop_signal.isSet():
+            time.sleep(1)
+    stop_signal.set()
```

Comment:
```python
for _ in range(wait_seconds):
    if stop_signal.isSet():
        break # don't keep looping if it's already set
    time.sleep(1)
else: # don't bother setting if we broke out of the loop (because it's already set)
    stop_signal.set()
```

Reply:
Actually, this whole looping/sleeping thing might go away if you use `Timer` instead (see comment below).

Reply:
Created healper for thread variant as it is simpler than `Timer` variant.

```def sleep_and_set_stop_signal_task(stop_signal, wait_seconds):
    """Busy waits for wait_seconds seconds before setting stop_signal."""
    def sleep_and_set_stop_signal_fn():
        for _ in range(4 * wait_seconds):
            if not stop_signal.isSet():
                time.sleep(0.25)
        stop_signal.set()

    thread = Thread(target=sleep_and_set_stop_signal_fn, args=())
    thread.daemon = True
    thread.start()


def sleep_and_set_stop_signal_timer(stop_signal, wait_seconds):
    """Busy waits for wait_seconds seconds before setting stop_signal."""
    limit = 4 * wait_seconds

    def sleep_and_set_stop_signal_fn(iteration):
        if iteration < limit:
            if not stop_signal.isSet():
                Timer(0.25, sleep_and_set_stop_signal_fn, args=(iteration + 1,)).start()
        else:
            stop_signal.set()

    Timer(0.25, sleep_and_set_stop_signal_fn, args=(0,)).start()
```

Reply:
This is what I was envisioning:
```python
def sleep_and_set_stop_signal_timer(stop_signal, wait_seconds):          
    """Busy waits for wait_seconds seconds before setting stop_signal."""
    t = Timer(4 * wait_seconds, stop_signal.set)                         
    t.daemon = True                                                      
    t.start()                                                            
```

Reply:
Oops, my bad. Done.